hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetUtils.java

```diff
@@ -229,6 +230,23 @@ public static InetSocketAddress createSocketAddr(String target,
     }
     target = target.trim();
     boolean hasScheme = target.contains("://");
+    if (NetUtils.isIPV6Address(target)) {
```

Comment:
NetUtils.isIPV6Address(target) is slow

<img width="1122" height="651" alt="image" src="https://github.com/user-attachments/assets/7432b57c-acee-48b3-9c26-6520b015a593" />